crates/subspace-networking/src/node_runner.rs

```diff
@@ -246,12 +246,12 @@ impl NodeRunner {
             SwarmEvent::ConnectionEstablished { .. } => {
                 self.shared
                     .connected_peers_count
-                    .fetch_add(1, Ordering::Relaxed);
+                    .fetch_add(1, Ordering::SeqCst);
             }
             SwarmEvent::ConnectionClosed { .. } => {
                 self.shared
                     .connected_peers_count
-                    .fetch_sub(1, Ordering::Relaxed);
+                    .fetch_sub(1, Ordering::SeqCst);
```

Comment:
`Relaxed` should be fine actually (if reading sometimes not up-to-date value is okay for us):
https://en.cppreference.com/w/cpp/atomic/memory_order#Relaxed_ordering

Reply:
>> They only guarantee atomicity and modification order consistency

`fetch_add` and `fetch_sub` are guaranteed to be atomic. Relaxed means that there are no guarantees in terms of order of those operation. They should happen eventually:
https://doc.rust-lang.org/nomicon/atomics.html#relaxed

Reply:
Exactly, and with increment/decrement we want that order guarantee or else `0-1+1` on unsigned type will not result in what you expect, hence `SeqCst`.